### PR TITLE
Update mypy to 0.931

### PR DESCRIPTION
**Pull request by nbrahms**
Also add a cast to work-around a poor type definition for the wcmatch
library.

Also add mypy to our dev packages to make for easier mypy debugging.

PR checklist:
- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -156,7 +156,6 @@ semgrep-core/src/cli/version.ml
 TAGS
 /semgrep-core/log_config.json
 /semgrep-core/.vscode/settings.json
-/Pipfile
```
Comment (aryx) [inline .gitignore:159]: oops

File: semgrep/semgrep/default_group.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 import click
 
+
```
Comment (aryx) [inline semgrep/semgrep/default_group.py:8]: change intended?

File: semgrep/semgrep/target_manager.py [hunk 3]
```diff
@@ -319,7 +320,12 @@ def filter_includes(
 
         includes = TargetManager.preprocess_path_patterns(includes)
         return frozenset(
-            wcglob.globfilter(arr, includes, flags=wcglob.GLOBSTAR | wcglob.DOTGLOB)
+            cast(
```
Comment (aryx) [inline semgrep/semgrep/target_manager.py:323]: maybe a comment here explaining why we need that? What is wrong with wcglob? Or do we need
to update the stub for this library?

**Review by aryx — APPROVED** [review]
Fantastic!
Thx a lot.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -34,7 +34,7 @@ repos:
       - id: mypy
         exclude: ^semgrep\/tests\/.+$|^setup.py$|^scripts\/.+$|^parsing-stats\/.+$|^perf\/.+$$
         args: [--config, mypy.ini, --show-error-codes]
-        additional_dependencies:
+        additional_dependencies: &mypy-deps
```
Comment (aryx) [inline .pre-commit-config.yaml:37]: ohoh, yaml anchor master!
Look at that @emjin !

**Review by aryx — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (nbrahms) [reply .gitignore:159]: This is intentional; otherwise you can't add the Pipfile changes

File: semgrep/semgrep/default_group.py (hunk 2 above)
Comment (nbrahms) [reply semgrep/semgrep/default_group.py:8]: This is from the black formatter

File: semgrep/semgrep/target_manager.py (hunk 3 above)
Comment (nbrahms) [reply semgrep/semgrep/target_manager.py:323]: We don't have control over the stub. I'll add a comment

File: .pre-commit-config.yaml (hunk 4 above)
Comment (nbrahms) [reply .pre-commit-config.yaml:37]: Tada

File: semgrep/semgrep/target_manager.py (hunk 3 above)
Comment (nbrahms) [reply semgrep/semgrep/target_manager.py:323]: ah lol the auto-merge went through before your comment

File: .gitignore (hunk 1 above)
Comment (aryx) [reply .gitignore:159]: I know, I meant why it was in the gitignore in the first place??

**Pull request merged**